niworkflows/interfaces/cifti.py

```diff
@@ -318,6 +318,12 @@ def _create_cifti_image(bold_file, label_file, bold_surfs, annotation_files, tr,
         warnings.warn("Resampling bold volume to match label dimensions")
         bold_img = resample_to_img(bold_img, label_img)
 
+    bold_orient = nb.aff2axcodes(bold_img.affine)
+    label_orient = nb.aff2axcodes(label_img.affine)
+    if bold_orient != label_orient:
+        bold_img = _reorient_image(bold_img, bold_orient, label_orient)
+    assert nb.aff2axcodes(bold_img) == label_orient == tuple('LAS')
```

Comment:
I would do the checks inside `_reorient_image`, so that you don't need to pass the orientations - just `bold_img` and `label_img`.